src/Databases/DatabaseReplicated.cpp

```diff
@@ -719,8 +730,9 @@ void DatabaseReplicated::recoverLostReplica(const ZooKeeperPtr & current_zookeep
             {
                 if (name != it->second)
                 {
+                    String intermediate_name = fmt::format("{}-{}", name, sipHash64(fmt::format("{}-{}", name, salt)));
```

Comment:
```suggestion
                    String intermediate_name = fmt::format(".rename-{}-{}", name, sipHash64(fmt::format("{}-{}", name, salt)));
```

Reply:
Also we can skip renaming to `intermediate` name if `from` name already starts with `".rename-"` and `to` name does not. Imagine that for some reason we constantly fail to do the second stage of renaming. We will append some random number on each iteration until we end up with extremely long table names.